lib/Target/Btor/BtorToBtorIRTranslation.cpp

```diff
@@ -0,0 +1,781 @@
+            auto opType = builder.getIntegerType( operandWidth );
+            assert( operandWidth > kids[1] && kids[1] >= kids[2] );
+            auto resType = builder.getIntegerType( kids[1] - kids[2] + 1 );
+
+            auto u = builder.create<btor::ConstantOp>(unknownLoc, opType, 
+                                                builder.getIntegerAttr(opType, kids[1]));
+            auto l = builder.create<btor::ConstantOp>(unknownLoc, opType, 
+                                                builder.getIntegerAttr(opType, kids[2]));
+            res = builder.create<btor::SliceOp>(unknownLoc, resType, cache.at( kids[0]), 
+                                                u->getResult(0), l->getResult(0));
+        }
+        break;
+        case BTOR2_TAG_sext: 
+            res = builder.create<btor::SExtOp>(unknownLoc, 
+                                            cache.at(kids[0]),
+                                            builder.getIntegerType( 
+                                                line->sort.bitvec.width));
+        break;
+        case BTOR2_TAG_uext: 
+            res = builder.create<btor::UExtOp>(unknownLoc, 
+                                            cache.at(kids[0]),
+                                            builder.getIntegerType( 
+                                                line->sort.bitvec.width));
+        break;
+
+        // ternary ops
+        case BTOR2_TAG_ite:
+            res = builder.create<btor::IteOp>(unknownLoc, 
+                                        cache.at(kids[0]),
+                                        cache.at(kids[1]),
+                                        cache.at(kids[2]));
+        break;
+
+        // unmapped ops
+        case BTOR2_TAG_read:
+        case BTOR2_TAG_constraint:
+        case BTOR2_TAG_init: 
+        case BTOR2_TAG_input:
+        case BTOR2_TAG_next:
+        case BTOR2_TAG_sort:
+        case BTOR2_TAG_state:
+        case BTOR2_TAG_fair:
+        case BTOR2_TAG_justice:
+        case BTOR2_TAG_output:
+        case BTOR2_TAG_write:
+        default:
+            break;
+    }
+    return res;
+}
+
+void cleanupCreatedLines() {
+    for( auto it = reached_lines.begin(); it != reached_lines.end(); ++it ) {
+        if( it->first < 0 ) {
+            Btor2Line *line = it->second;
+            delete [] line->args;
+            delete line;
+        }
+    }
+}
+
+void createNegateLine( int64_t curAt, Btor2Line * line ) {
+    Btor2Line *res = new Btor2Line;
+    int64_t *args = new int64_t[1];
+    args[0] = curAt * -1;
+
+    res->id = curAt;
+    res->lineno = curAt;
+    res->name = "not";
+    res->tag = BTOR2_TAG_not;
+    res->sort = line->sort;
+    res->init = line->init;
+    res->next = line->next;
+    res->constant = line->constant;
+    res->symbol = line->symbol;
+    res->nargs = 1;
+    res->args = args; 
+
+    reached_lines[ curAt ] = res;
+}
+
+bool isValidChild( int64_t line ) {
```

Comment:
functions that take untyped integer arguments are often abused. I would use `Btor2Line` instead and faction our the lookup to call site

Reply:
Sounds good